### ISSUE TITLE
NullPointerException at vgscollect.view.InputFieldView.findFocus

### ISSUE THREAD

**Issue by mikkoville** [verygoodsecurity/vgs-collect-android#259]
**Describe the bug**
After upgrading to latest version VGS `1.6.14` we have been able to reproduce nullpointers related:
`com.verygoodsecurity.vgscollect.view.InputFieldView.findFocus`

Full stacktrace:
```
2021-10-04 14:03:47.279 3813-3813/com.xxxx.android.master E/MessageQueue-JNI: java.lang.NullPointerException: inputField.findFocus() must not be null
        at com.verygoodsecurity.vgscollect.view.InputFieldView.findFocus(InputFieldView.kt:480)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewGroup.findFocus(ViewGroup.java:1231)
        at android.view.ViewRootImpl.focusableViewAvailable(ViewRootImpl.java:4119)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.ViewGroup.focusableViewAvailable(ViewGroup.java:940)
        at android.view.View.setFlags(View.java:15181)
        at android.view.View.setVisibility(View.java:10836)
        at com.xxxx.android.core_ui.utils.UiExtensionsKt.setVisible(UiExtensions.kt:58)
        at com.xxxx.android.core_ui.widget.InlineNotificationWidget.show(InlineNotificationWidget.kt:106)
        at com.xxxx.android.payment.controllers.add_card.AddCardController.setCardInputErrors(AddCardController.kt:234)
        at com.xxxx.android.payment.controllers.add_card.AddCardController.onCardInputInvalid(AddCardController.kt:173)
        at com.xxxx.android.payment.payment_services.VgsWrapper.handleFieldStateChange(VgsWrapper.kt:152)
        at com.xxxx.android.payment.payment_services.VgsWrapper.access$handleFieldStateChange(VgsWrapper.kt:37)
        at com.xxxx.android.payment.payment_services.VgsWrapper$onInflate$1.onStateChange(VgsWrapper.kt:98)
        at com.verygoodsecurity.vgscollect.core.storage.content.field.TemporaryFieldsStorage.notifyUserOnDataChange(TemporaryFieldsStorage.kt:112)
        at com.verygoodsecurity.vgscollect.core.storage.content.field.TemporaryFieldsStorage.processNewState(TemporaryFieldsStorage.kt:76)
        at com.verygoodsecurity.vgscollect.core.storage.content.field.TemporaryFieldsStorage.addItem(TemporaryFieldsStorage.kt:60)
        at com.verygoodsecurity.vgscollect.core.storage.content.field.TemporaryFieldsStorage$performSubscription$1.emit(TemporaryFieldsStorage.kt:54)

```

**To Reproduce**
1. We listen for changes in `vgsCollect.addOnFieldStateChangeListener`
2. Based on if `FieldState.isValid` we hide one view in the same parent `ViewGroup` that `VGSCardNumberEditText` exists. Note this view that we hide or show (set to View.VISIBLE or View.GONE) is not a from VGS collect. It is a normal Android `ConstraintLayout`(InlineNotificationWidget in the stack) that belongs to the same parent `ViewGroup` the VGS collect views exist.
3. Get null pointer like in the stack trace above

For the parent `ViewGroup` that contains VGS collect views and our other `ConstraintLayout` we have set:
```
isFocusable = true
isFocusableInTouchMode = true
```
**Expected behavior**
Should not crash the application

**Screenshots**

**Environment (please complete the following information):**
 - SDK Version [1.6.14]
 - Android Version: [10]
 - Device: [Pixel 5 emulator]

**Additional context**

**Possible Solution**
If you have the solution on how to fix the issue.

**Comment by DmytroDm** [comment]
@mikkoville Thank you for informing us about this issue. We will have a look at this.

**Comment by DmytroDm** [comment]
@mikkoville We had released a new version `1.6.15` with a bug fix.

**Comment by mikkoville** [comment]
@DmytroDm verified. Does not crash anymore. Thanks 👍
